src/main/java/httpServer/http/request/RequestBuilder.java

```diff
@@ -0,0 +1,81 @@
+package httpServer.http.request;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static httpServer.http.Constants.crlf;
+
+public class RequestBuilder {
+
+    public String method;
+    public String resource;
+    public String httpVersionNumber;
+    public Map<String, String> headers = new HashMap<>();
+    public String body;
+
+    public RequestBuilder buildRequestLine(String line) {
+        String[] requestLine = line.split(" ");
+        this.method = requestLine[0];
+        this.resource = requestLine[1];
+        this.httpVersionNumber = requestLine[2].replace(crlf, "");;
+        return this;
+    }
+
+    public RequestBuilder buildHeaderLine(String line) {
+        String[] requestLine = line.split(": ");
+        String removedCLDR = requestLine[1].replace(crlf, "");
+        this.headers.put(requestLine[0], removedCLDR);
+        return this;
+    }
+
+    public RequestBuilder setBody(String line) {
+        this.body = line;
+        return this;
+    }
+
+    public String toString() {
+        String requestLine = method + resource + httpVersionNumber + crlf;
+        System.out.println(requestLine);
+        headers.forEach((k, v) -> System.out.println("key: " + k + ", value: " + v));
+        if (body != null) {
+            System.out.println("body: " + body);
+        }
```

Comment:
Are some of these for debugging purposes?

Reply:
the ToString function itself is for debugging purposes... I don't want to dismantle it entirely just yet though, in case I need it

Reply:
If so, then you'll need to write tests for it, including both test cases for with and without a body, single header, multiple headers, etc.

Reply:
I deleted it because the testing seemed like a bad use of my time and I wasn't using it anymore!